### ISSUE TITLE
Support vim word completion with `ctrl-p` and `ctrl-n`

### ISSUE THREAD

**Issue by quinncomendant** [zed-industries/zed#17461]
### Check for existing issues

- [X] Completed

### Describe the feature

It doesn't seem possible to do [vim word completion](https://vimdoc.sourceforge.net/htmldoc/usr_24.html#24.3) in Zed. In regular vim, it's `ctrl-p`/`ctrl-n`, but in Zed that goes to previous line/next line.

For example, if I type the following, then hit `ctrl-p` after typing `autoc`, vim mode should complete the word to `autocomplete`:

```
This document contains the word autocomplete.
Let’s try to autoc<ctrl-p>
```

Expected result:

```
This document contains the word autocomplete.
Let’s try to autocomplete
```

VIM autocomplete has a lot of features that would be useful in Zed. The docs include some useful options:

> The Vim editor goes through a lot of effort to find words to complete.  By default, it searches the following places:
> 
> 	1. Current file
> 	2. Files in other [windows](https://vimdoc.sourceforge.net/htmldoc/windows.html#windows)
> 	3. Other loaded files (hidden [buffers](https://vimdoc.sourceforge.net/htmldoc/windows.html#buffers))
> 	4. Files which are not loaded (inactive [buffers](https://vimdoc.sourceforge.net/htmldoc/windows.html#buffers))
> 	5. Tag files
> 	6. All files #included by the current file

…

> If you know what you are looking for, you can use these commands to complete with a certain type of item:
> 
> 	[CTRL-X](https://vimdoc.sourceforge.net/htmldoc/change.html#CTRL-X) [CTRL-F](https://vimdoc.sourceforge.net/htmldoc/scroll.html#CTRL-F)		file names
> 	[CTRL-X](https://vimdoc.sourceforge.net/htmldoc/change.html#CTRL-X) [CTRL-L](https://vimdoc.sourceforge.net/htmldoc/various.html#CTRL-L)		whole lines
> 	[CTRL-X](https://vimdoc.sourceforge.net/htmldoc/change.html#CTRL-X) [CTRL-D](https://vimdoc.sourceforge.net/htmldoc/scroll.html#CTRL-D)		[macro](https://vimdoc.sourceforge.net/htmldoc/map.html#macro) [definitions](https://vimdoc.sourceforge.net/htmldoc/intro.html#definitions) (also in included files)
> 	[CTRL-X](https://vimdoc.sourceforge.net/htmldoc/change.html#CTRL-X) [CTRL-I](https://vimdoc.sourceforge.net/htmldoc/motion.html#CTRL-I)		current and included files
> 	[CTRL-X](https://vimdoc.sourceforge.net/htmldoc/change.html#CTRL-X) CTRL-K		words from a dictionary
> 	[CTRL-X](https://vimdoc.sourceforge.net/htmldoc/change.html#CTRL-X) [CTRL-T](https://vimdoc.sourceforge.net/htmldoc/tagsrch.html#CTRL-T)		words from a thesaurus
> 	[CTRL-X](https://vimdoc.sourceforge.net/htmldoc/change.html#CTRL-X) [CTRL-\]](https://vimdoc.sourceforge.net/htmldoc/tagsrch.html#CTRL-\])		[tags](https://vimdoc.sourceforge.net/htmldoc/tagsrch.html#tags)
> 	[CTRL-X](https://vimdoc.sourceforge.net/htmldoc/change.html#CTRL-X) [CTRL-V](https://vimdoc.sourceforge.net/htmldoc/visual.html#CTRL-V)		Vim command line

**Comment by kjyv** [comment]
It seems to work fine now, there is a popup with completions in 0.196.6

**Comment by quinncomendant** [comment]
Yes, Zed vim mode now supports basic autocomplete for words in the same document.

Should we keep this ticket open as a feature request for the full suite of VIM autocomplete options (e.g., [`CTRL-X`](https://vimdoc.sourceforge.net/htmldoc/change.html#CTRL-X) [`CTRL-F`](https://vimdoc.sourceforge.net/htmldoc/scroll.html#CTRL-F) to complete file names, etc – see the full list above)?

**Comment by stephenh** [comment]
Fwiw I'd like to have a binding that doesn't "show the popup", I want a binding that "just assumes the 1st entry in the popup is correct, and uses that" (but then be able to cycle to the next suggestion by hitting the binding again).

I.e. with the current behavior: `autoc<ctrl-p>` I have to:

* Type `ctrl-p`
* See the popup
* Press enter to select `autocomplete`
* The rest of `omplete` is entered

I'd like to just:

* Type `ctrl-p`
* Have `omplete` entered

If this 1st entry-from-the-popup was correct, I can move on.
